diff --git a/src/JSDOM.hs b/src/JSDOM.hs
index debdb4d..2731c46 100644
--- a/src/JSDOM.hs
+++ b/src/JSDOM.hs
@@ -67,26 +67,7 @@ inAnimationFrame :: OnBlocked       -- ^ what to do when encountering a blocking
                  -> (Double -> JSM ())  -- ^ the action to run
                  -> JSM AnimationFrameHandle
 inAnimationFrame _ f = do
-    handlersMVar <- animationFrameHandlers <$> askJSM
-    -- Take the list of pending animation fram handlers
-    handlers <- liftIO $ takeMVar handlersMVar
-    -- Add this handler to the list to be run by the callback
-    liftIO $ putMVar handlersMVar (f : handlers)
-    -- If this was the first handler added set up a callback
-    -- to run the handlers in the next animation frame.
-    when (null handlers) $ do
-        win <- currentWindowUnchecked
-        rec cb@(RequestAnimationFrameCallback (Callback fCb)) <- newRequestAnimationFrameCallbackSync $ \t -> do
-              -- This is a one off handler so free it when it runs
-              freeFunction fCb
-              -- Take the list of handers and empty it
-              handlersToRun <- liftIO $ takeMVar handlersMVar
-              liftIO $ putMVar handlersMVar []
-              -- Exectute handlers in the order 
-              forM_ (reverse handlersToRun) (\handler -> handler t)
-        -- Add the callback function
-        void $ requestAnimationFrame win cb
-    return AnimationFrameHandle
+  error "inAnimationFrame NYI for experimental-core2"
 
 #endif
 
